src/chunk.c

```diff
@@ -3411,44 +3334,59 @@ chunk_update_status(FormData_chunk *form)
 								   BTEqualStrategyNumber,
 								   F_INT4EQ,
 								   Int32GetDatum(chunk_id));
+	bool success = false;
+	bool dropped, dropped_isnull;
 
 	ts_scanner_foreach(&iterator)
 	{
 		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
-		bool dropped_isnull, status_isnull;
-		int status;
 
 		dropped = DatumGetBool(slot_getattr(ti->slot, Anum_chunk_dropped, &dropped_isnull));
 		Assert(!dropped_isnull);
-		status = DatumGetInt32(slot_getattr(ti->slot, Anum_chunk_status, &status_isnull));
-		Assert(!status_isnull);
-		if (!dropped && status != new_status)
+		if (!dropped)
 		{
-			success = chunk_update_status_internal(form); // get RowExclusiveLock and update here
+			ts_chunk_formdata_fill(form, ti);
+			success = true;
+			break;
 		}
 	}
 	ts_scan_iterator_close(&iterator);
-	if (dropped)
-		ereport(ERROR,
-				(errcode(ERRCODE_INTERNAL_ERROR),
-				 errmsg("attempt to update status(%d) on dropped chunk %d", new_status, chunk_id)));
+
 	return success;
 }
 
+/* Chunk catalog updates are done in two steps.
```

Comment:
This comment should move to before chunk_update_catalog_tuple.
would be good to add dire warnings  about concurrency issues if you fail to do so :-)